File: Java/PCA9536.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// PCA9536
// This code is designed to work with the PCA9536_I2CIO I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/content/Digital-IO?sku=PCA9536_I2CIO#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class PCA9536
{
	public static void main(String args[]) throws Exception
	{
		// Create I2C bus
		I2CBus Bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, PCA9536 I2C address is 0x41(65)
		I2CDevice device = Bus.getDevice(0x41);

		// Select configuration register
		// All pins configured as inputs
		device.write(0x03, (byte)0xFF);

		// Output to screen
		System.out.printf("All Pins State are HIGH %n");
		Thread.sleep(500);

		// Read 1 byte of data
		byte[] data = new byte[1];
		data[0] = (byte)device.read(0x00);

		// Convert the data to 4-bits
		int data1 = (data[0] & 0x0F);

		for(int i=0; i<4; i++)
		{
			if((data1 & ((int)Math.pow(2, i))) == 0)
			{
				System.out.printf("I/O Pin %d State is LOW %n", i);
			}
			else
			{
				System.out.printf("I/O Pin %d State is HIGH %n", i);
				Thread.sleep(500);
			}
		}
	}
}

File: Python/PCA9536.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# PCA9536
# This code is designed to work with the PCA9536_I2CIO I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/content/Digital-IO?sku=PCA9536_I2CIO#tabs-0-product_tabset-2

import smbus
import time

# Get I2C bus
bus = smbus.SMBus(1)

# PCA9536 address, 0x41(65)
# Select configuration register, 0x03(03)
#		0xFF(255)	All pins configured as inputs
bus.write_byte_data(0x41, 0x03, 0xFF)

# Output to screen
print "All Pins State are HIGH"

time.sleep(0.5)

# PCA9536 address, 0x41(65)
# Read data back from 0x00(00), 1 byte
data = bus.read_byte_data(0x41, 0x00)

# Convert the data to 4-bits
data = (data & 0x0F)

for i in range(0, 4) :
	if (data & (2 ** i)) == 0 :
		print "I/O Pin %d State is LOW" %i
	else :
		print "I/O Pin %d State is HIGH" %i
		time.sleep(0.5)
